kratos/solving_strategies/builder_and_solvers/residualbased_elimination_builder_and_solver_with_constraints.h

```diff
@@ -380,6 +372,28 @@ class ResidualBasedEliminationBuilderAndSolverWithConstraints
         KRATOS_CATCH("ResidualBasedEliminationBuilderAndSolverWithConstraints failed to finalize solution step.")
     }
 
+    /**
+     * @brief This method provides the defaults parameters to avoid conflicts between the different constructors
+     * @return The default parameters
+     */
+    const Parameters GetDefaultParameters() const override
+    {
+        Parameters class_default_parameters = Parameters(R"(
+        {
+            "name"                                 : "elimination_builder_and_solver_with_constraints",
+            "check_constraint_relation"            : true,
+            "reset_relation_matrix_each_iteration" : true
+        })");
+
+        // Getting base class default parameters
+        const Parameters base_default_parameters = BaseType::GetDefaultParameters();
+        class_default_parameters.RecursivelyAddMissingParameters(base_default_parameters);
+
+        const Parameters default_parameters(class_default_parameters);
```

Comment:
why the duplicate?

Ah because it returns a const object

I would change the signature, I don't think it is necessary to return a const object here, kinda doesn't make sense

Reply:
Ok